py_experimenter/experimenter.py

```diff
@@ -203,14 +332,36 @@ def execution_wrapper(self, approach, custom_fields: dict, keyfields, result_pro
             else:
                 result_processor._change_status('done')
 
-    def reset_experiments(self, status= 'error') -> None:        
+    def reset_experiments(self, status= 'error') -> None:
+        """
+        Deletes the experiments of the database table having the given status and adds the according keyfields as new rows.
+        
+        * `created` when the experiment is added to the database table, execution has not started.
+        * `running` when the execution of the experiment has been started.
+        * `error` if something went wrong during the execution, i.e., an exception is raised
+        * `done` if the execution finished successfully.
+
+        :param status: The status of experiments that should be reset. Defaults to 'error'.
+        :type status: str, optional
+        """
         keyfields,entries = self._dbconnector.delete_experiments_with_status(status)
         rows = self._extract_row_from_entries(keyfields, entries)
         if rows:
             self.fill_table_with_rows(rows)
         logging.info(f"{len(rows)} experiments with status {status} were reset")
         
-    def _extract_row_from_entries(self, keyfields, entries):
+    def _extract_row_from_entries(self, keyfields: List[str], entries: List[List[str]]) -> List[dict]:
+        """
+        Creates a list of rows based on the given `keyfields` having values as defined in the given `entries`. 
```

Comment:
I am a little confused by this description. A list of dicts is returned, which I am not sure I would describe as rows except if this refers to mappings for database rows? Maybe the doc could start with 'convenience function for' and state the purpose?

Reply:
Moved this method as internal one within `reset_experiments` as it is only used there. Additionally, renamed variables and method name to be more meaningful. Furthermore, removed code documentation for that method, as it is only an internal one.